I5 setup storybook js

ThePhisch opened the pull request:
Installed storybook and ported the first story components. Installed further dependencies including `@primer/css`, `bulma`, `sass`, `sass-loader` as needed.

Commit:
Install Storybook

Installed Storybook with webpack5 as builder. Also added `eslint-plugin`
and did the `npm7` migration (generating the `.npmrc`-file), as
recommended by the storybook installer.

Deleted the contents of `src/stories`, meaning that Storybook has no
stories it can display. The server starts up though.

Commit:
Ported `Alert`, `Box`, `Button`, `Separator` stories

The stories were ported from the old webui. The stories were chosen
since they only rely on the `primer` CSS-library. Also ported the
`rucio.d.ts` file containing the rucio webui story components proptype
definitions.

In the process, sass support was activated for the Storybook server. In
addition, `@primer/css` was installed (as a depency for the story).

maany commented on .npmrc:
```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```
what package requires the `legacy-peer-deps` flag to be enabled?

ThePhisch replied:
I was prompted to add this flag during the installation of `storybook`. It seems to be a [known problem](https://github.com/storybookjs/storybook/issues/18298).

maany reviewed (changes requested):
Looks good! If we can avoid using `legacy-peer-deps` it would be best. But for some packages, it is unavoidable.
Also, can we move the stories inside the `src` directory? and rename it to `component-library`?

ThePhisch commented:
> Also, can we move the stories inside the `src` directory? and rename it to `component-library`?

I will rename `src/stories` to `src/component-library`, as discussed.

Commit:
Port Simple Components Including Bulma

Ported components (stories) into `src/component-library`. These
components are called *simple* because they use Bulma CSS (and related
dependencies) but do not rely on other components. In addition, if a
component requires the handling of images, it is omitted in this commit.

Added Components:

* Checkbox (requires `bulma-checkradio`)
* Dropdown
* Form
* Input
* Modal
* ProgressBar
* Steps (requires `bulma-steps`)
* Table
* Tabs
* ToggleSwitch (requires `bulma-switch`)

maany reviewed (changes requested):
Can we already migrate to Storybook 7
https://chromatic-ui.notion.site/Storybook-7-migration-guide-dbf41fa347304eb2a5e9c69b34503937 

Also, can you refactor the component-library directory to have 2 sub-directories, one for components and one for sass/scss

The components can import from the sass/scss

Commit:
Changed file structure

Switched to a system in which `src/component-library` has contents:

* `components` which are the story components that were previously
  directly under `component-library`
* `scss` containing stylesheets important for the stories. The stories
  import the `main.scss`-file.

The reasoning behind this is that each component should work in
isolation and its styles should not depend on globally imported
stylesheets.

Commit:
Upgrade to Storybook v7 (beta)

Moved to Storybook `7.0.0-beta.49`. Had to add `util` to `resolve.fallback`, installed `util`.

In upgrading, accepted all suggested migrations except for `autodocs:
true`. These are:
* `storybook-binary`
* `sb-scripts`
* `newFrameworks` (which removes webpack5)
* `nextjsFramework` (which replaces `newFrameworks` with one specific to
  `nextjs`
* `bare-mdx-stories-glob`

Commit:
Redid PrimerCSS-Components in TailwindCSS

After deciding to not follow up on using PrimerCSS for many of the
designs (due to the persisting issue of not being able to compile the
SCSS files), the components taking style directions from PrimerCSS were
redone to make use of TailwindCSS.

In redoing the components, the interfaces found under `rucio.d.ts` were
also changed. These changes primarily consisted of simpifying the
interfaces to a minimum. In some cases, such as `CheckboxProps`, new
parameters were added (`CheckboxProps` now has the optional attribute
`disabled` so that the behaviour of other buttons is matched).

The rework touched the Javascript logic in some cases, but mainly tried
to port the designs from PrimerCSS into TailwindCSS. The components
redone are:

* Alert (split into Standard, Warning, Success and Error)
* Box (split into Condensed, Spacious, Blue, Danger)
* Button (Button, Checkbox, Radiobutton) (Switch remains unchanged)

Commit:
Work on Tailwind: Input

Ported the `input` component into tailwind. Split into two components
for numeric and text input.

Also removed unnecessary scss files and moved `rucio.d.ts` to
`src/component-library/components.d.ts` to reflect that this file only
provides the interfaces for the components.

maany commented:
can you rebase?
